runbook/kubeone_ip_change.md

```diff
@@ -1,90 +1,164 @@
-```
-openssl x509 -in /etc/kubernetes/pki/apiserver.crt -text
-```
-Look for the `X509v3 Subject Alternative Name` line, after which will be a list of all the DNS names and IP addresses that are included on the certificate as SANs. After following this procedure, you should see the newly-added names and IP addresses you specified in the modified kubeadm configuration file. If you don’t, then something went wrong along the way. Common mistakes in this process include forgetting to remove the previous certificate and key (kubeadm won’t create new ones if they already exist), or failing to include the --config kubeadm-config.yaml on the kubeadm init phase certs command.
+    ```
+    kubectl edit -n kube-public cm kube-proxy
 
-## Restarting API server
+    ```
 
-The final step is restarting the API server to pick up the new certificate. The easiest way to do this is to kill the API server container using docker:
+    - Update seed.yaml with new `KUBECONFIG` (with base64 encoding)
 
-Run `docker ps | grep kube-apiserver | grep -v pause` to get the container ID for the container running the Kubernetes API server. (The container ID will be the very first field in the output.)
+    ```
+    cat xxxxx-kubeconfig | base64 | tr -d '\n\r'
 
-Run `docker kill <containerID>` to kill the container.
+    ```
 
-If your nodes are running containerd as the container runtime, the commands are a bit different:
+  - Add the new encoded kubeconfig in seed.yaml as below
 
-Run `crictl pods | grep kube-apiserver | cut -d' ' -f1` to get the Pod ID for the Kubernetes API server Pod.
+    ```
+    apiVersion: v1
+    kind: Secret
+    metadata:
+    name: xxxx-kubeconfig
+    namespace: kubermatic
+    type: Opaque
+    data:
+    kubeconfig: <add base64 encoded kubeconfig>
+    <truncated>
+    ...
 
-Run `crictl stop <pod-id>` to stop the Pod.
+    ```
 
-Run `crictl rmp <pod-id>` to remove the Pod.
+  - Apply the new seed.yaml
 
-The Kubelet will automatically restart the container, which will pick up the new certificate. As soon as the API server restarts, you will immediately be able to connect to it using one of the newly-added IP addresses/DNS names.
+    ```
+    kubectl apply -f seed.yaml
 
-## Updating the In-Cluster Configuration
+    ```
 
-Assuming everything is working as expected, the final step is to update the kubeadm ConfigMap stored in the cluster. This is important so that when you use kubeadm to orchestrate a cluster upgrade later, the updated information will be present in the cluster.
+  - Verify `/var/lib/kubelet/kubeconfig` on all master servers for its contents and ensure it reflects the new values of cluster endpoint.
 
-```
-kubeadm init phase upload-config kubeadm --config kubeadm-config.yaml
-```
+Once all the above metioned activities are completed you will be able to see all the master/worker nodes in ready state and all pods in running state as well. 
```

Comment:
I wounder if this changes are enough for a persistent change of a kubeone cluster. I think you need to update also the `kubeone.yaml` to the new API endpoint, otherwise a upgrade could potential revert the change.
Could you test the upgrade of a new kubernetes version and pot. add missing parts?

Reply:
will test with new kubernetes version and update the runbook. Once completed will raise a new PR.